tests/phpunit/CRM/Utils/AddressTest.php

```diff
@@ -43,33 +43,33 @@ public function testStateProvinceFormattedBillingAddress() {
       'country-99' => 'United States',
     ];
 
-    // set address_format (we are only interested in state_province & state_province_name)
+    // Set address_format (we are only interested in state_province & state_province_name).
     $addFormat = '{contact.state_province}';
     Civi::settings()->set('address_format', $addFormat);
     $formatted_address = CRM_Utils_Address::getFormattedBillingAddressFieldsFromParameters($params, '99');
-    $this->assertTrue((bool) $formatted_address == 'AL');
+    $this->assertEquals('AL', $formatted_address);
```

Comment:
so there seems to be trailing new line that is showing up here somehow @eileenmcnaughton

Reply:
I guess that's what formatted means...

And hence the previous code's (bool) to cast it true which then matched the string which autocasts to true so it always passed. How else do you deal with trailing newlines?

Reply:
@demeritcowboy so we could use that pattern to make all our tests pass....

Reply:
Or even simpler use the technique @mlutfy [discovered](https://github.com/civicrm/civicrm-core/pull/16838#issuecomment-659788833): Automatic pass!

Reply:
Another winning technique

Reply:
Hah - not sure even @mlutfy's trick woudl have prevented this one

![image](https://user-images.githubusercontent.com/336308/198187318-f633c148-bf08-4b9f-9d84-ea142396c0f0.png)

test this please